### PR TITLE
create vpg.py and added notes about VPG

### PR DESCRIPTION
**Pull request by vovw**

### CODE REVIEW

File: Tutorials/VPG_tutorial.md [hunk 1]
```diff
@@ -1 +1,69 @@
...
+        return torch.softmax(self.fc(x), dim=-1)
+
+def select_action(state):
+    state = torch.FloatTensor(state).unsqueeze(0)
+    probs = policy_net(state)
+    m = Categorical(probs)
+    action = m.sample()
+    return action.item(), m.log_prob(action)
+
+def update_policy(log_probs, rewards):
```
Comment (aPR0T0) [inline Tutorials/VPG_tutorial.md:176]: Explain update policy with mathematical counter parts

File: Tutorials/code/vpg.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        return torch.softmax(self.fc(x), dim=-1)
+
+def select_action(state):
+    state = torch.FloatTensor(state).unsqueeze(0)
+    probs = policy_net(state)
+    m = Categorical(probs)
+    action = m.sample()
+    return action.item(), m.log_prob(action)
+
+def update_policy(log_probs, rewards):
```
Comment (aPR0T0) [inline Tutorials/code/vpg.py:108]: Looks good just need to add more comments for each defined variable

Also, a gif would be better for the agents

File: Tutorials/VPG_tutorial.md [hunk 3]
```diff
@@ -1 +1,69 @@
 # Vanilla Policy gradient implementation
+
+
```
Comment (aPR0T0) [inline Tutorials/VPG_tutorial.md:5]: Define the simulation environment briefly and then introduce what is VPG and all and mention the cost function that we need to optimize

**Review by aPR0T0 — COMMENTED** [review]
LGTM, documentation needed

**Comment by vovw** [comment]
updated, not sure why 2 equations don't get properly rendered in GitHub, checked using pandoc it looks fine

**Comment by aPR0T0** [comment]
Changes needed:
1. Add an explanation for the cost function
2. Explain the algorithm in detail steps in between cost function and change in that cost function:
image

**Comment by vovw** [comment]
what do you think?

Screenshot 2024-09-18 at 4 23 24 AM

Screenshot 2024-09-18 at 4 21 42 AM

**Comment by aPR0T0** [comment]
LGTM!, you didn't explain policy gradient theorem though... make sure to include that as well

**Pull request merged**